File: p7.c
#include <stdio.h>

int isprime(int c)
{
    int i;
    if(c%2 == 0)
        return 0;

    for(i = 3; i*i <= c; i += 2)
        if(c%i == 0)
            return 0;

    return 1;
}


int main(void)
{
    int count = 1, i;

    printf("test: %d\n", isprime(104744));

    for(i = 3; count < 10001; i +=1)
        if(isprime(i))
            count += 1;

    printf("10001st prime is %d\n", i-1);

}

File: go/p7.go
package main

import "fmt"

func divides(a, b int) bool {
    return a%b == 0
}

func isPrime(x int) bool {
    if divides(x, 2) {
        return false
    }

    for i := 3; i*i <= x; i += 2 {
        if divides(x, i) {
            return false
        }
    }

    return true
}



func main() {

    count, i := 1, 3

    for {
        if isPrime(i) {
            count += 1
        }

        if count == 10001 {
            break
        } else {
            i += 2
        }
    }

    fmt.Println(i)
}

